From f27b194e2be446a59150cddd3b4b0277542dd2d6 Mon Sep 17 00:00:00 2001
From: Paul Gortmaker <paul.gortmaker@windriver.com>
Date: Mon, 9 May 2016 14:40:27 -0400
Subject: [PATCH 368/447] ACPI / GED: make evged.c explicitly non-modular

Git-commit: 437014bdac9680e3b32c006635ac270b808ce476
Patch-mainline: v4.7-rc1
References: fate#320512

The Makefile/Kconfig currently controlling compilation of this code is:

Makefile:acpi-$(CONFIG_ACPI_REDUCED_HARDWARE_ONLY) += evged.o

drivers/acpi/Kconfig:config ACPI_REDUCED_HARDWARE_ONLY
drivers/acpi/Kconfig:   bool "Hardware-reduced ACPI support only" if EXPERT

...meaning that it currently is not being built as a module by anyone.

Lets remove the couple traces of modularity so that when reading the
code there is no doubt it is builtin-only.

Since module_platform_driver() uses the same init level priority as
builtin_platform_driver() the init ordering remains unchanged with
this commit.

The file wasn't explicitly including the module.h file but it did
already have init.h so, unlike similar changes, this one has no
header changes at all.

We also delete the MODULE_LICENSE tag since all that information
is already contained at the top of the file in the comments.

Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 drivers/acpi/evged.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/acpi/evged.c b/drivers/acpi/evged.c
index 9c0a868..46f0603 100644
--- a/drivers/acpi/evged.c
+++ b/drivers/acpi/evged.c
@@ -151,6 +151,4 @@ static struct platform_driver ged_driver = {
 		.acpi_match_table = ACPI_PTR(ged_acpi_ids),
 	},
 };
-
-module_platform_driver(ged_driver);
-MODULE_LICENSE("GPL v2");
+builtin_platform_driver(ged_driver);
-- 
2.11.0

